### PR TITLE
feat(uat): add steps for tx and rx content type and extend t102

### PR DESCRIPTION
**Pull request by auarbekov**
**Issue #, if available:**
https://klika-tech.atlassian.net/browse/GGMQ-190
https://klika-tech.atlassian.net/browse/GGMQ-189
add tx/rx content type and extend t102 with content type

**Description of changes:**
- add content type steps
- extend t102 with content type
- rename t101
- rename t102

**Why is this change necessary:**
to implement Mqtt 5.0 features

**How was this change tested:**
Scenarios run manually and on CI

**Test results:**
```
Given my device is registered as a Thing....................................passed
And my device is running Greengrass.........................................passed
When I create a Greengrass deployment with components.......................passed
And I create client device "publisher"......................................passed
And I create client device "subscriber".....................................passed
When I associate "subscriber" with ggc......................................passed
When I associate "publisher" with ggc.......................................passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:.passed
And I update my Greengrass deployment configuration, setting the component aws.greengrass.client.Mqtt5JavaPahoClient configuration to:.passed
And I deploy the Greengrass deployment configuration........................passed
Then the Greengrass deployment is COMPLETED on the device after 5 minutes...passed
And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes.passed
Then I discover core device broker as "localMqttBroker1" from "publisher" in OTF.passed
Then I discover core device broker as "localMqttBroker2" from "subscriber" in OTF.passed
And I connect device "publisher" on aws.greengrass.client.Mqtt5JavaPahoClient to "localMqttBroker1" using mqtt "v5".passed
And I connect device "subscriber" on aws.greengrass.client.Mqtt5JavaPahoClient to "localMqttBroker2" using mqtt "v5".passed
And I set MQTT publish 'retain' flag to true................................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "receive_last_retain_message" with qos 0 and message "First retained message".passed
When I publish from "publisher" to "receive_last_retain_message" with qos 0 and message "Second retained message".passed
When I subscribe "subscriber" to "receive_last_retain_message" with qos 0...passed
And message "First retained message" is not received on "subscriber" from "receive_last_retain_message" topic within 5 seconds.passed
And message "Second retained message" received on "subscriber" from "receive_last_retain_message" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to true................................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "receive_only_retain_message_on_subscribe" with qos 0 and message "First message with retain".passed
And I set MQTT publish 'retain' flag to false...............................passed
When I publish from "publisher" to "receive_only_retain_message_on_subscribe" with qos 0 and message "Second message without retain".passed
When I subscribe "subscriber" to "receive_only_retain_message_on_subscribe" with qos 0.passed
And message "Second message without retain" is not received on "subscriber" from "receive_only_retain_message_on_subscribe" topic within 5 seconds.passed
And message "First message with retain" received on "subscriber" from "receive_only_retain_message_on_subscribe" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to true................................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "subscribe_twice_with_send_at_subscription" with qos 0 and message "Single message in case3".passed
When I subscribe "subscriber" to "subscribe_twice_with_send_at_subscription" with qos 0.passed
And message "Single message in case3" received on "subscriber" from "subscribe_twice_with_send_at_subscription" topic within 5 seconds.passed
And I clear message storage.................................................passed
When I subscribe "subscriber" to "subscribe_twice_with_send_at_subscription" with qos 0.passed
And message "Single message in case3" received on "subscriber" from "subscribe_twice_with_send_at_subscription" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to true................................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_NEW_SUBSCRIPTION".passed
When I publish from "publisher" to "send_at_new_subscription" with qos 1 and message "Single retained message in case4" and expect status 16.passed
When I subscribe "subscriber" to "send_at_new_subscription" with qos 0......passed
And message "Single retained message in case4" received on "subscriber" from "send_at_new_subscription" topic within 5 seconds.passed
And I clear message storage.................................................passed
When I subscribe "subscriber" to "send_at_new_subscription" with qos 0......passed
And message "Single retained message in case4" is not received on "subscriber" from "send_at_new_subscription" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to true................................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_DO_NOT_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "do_not_send_at_subscription" with qos 0 and message "Single retained message in case5".passed
When I subscribe "subscriber" to "do_not_send_at_subscription" with qos 0...passed
And message "Single retained message in case5" is not received on "subscriber" from "do_not_send_at_subscription" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to false...............................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "t102_case6" with qos 0 and message "Single not retained message in case6".passed
When I subscribe "subscriber" to "t102_case6" with qos 0....................passed
And message "Single not retained message in case6" is not received on "subscriber" from "t102_case6" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to false...............................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_SEND_AT_NEW_SUBSCRIPTION".passed
When I publish from "publisher" to "t102_case7" with qos 0 and message "Single not retained message in case7".passed
When I subscribe "subscriber" to "t102_case7" with qos 0....................passed
And message "Single not retained message in case7" is not received on "subscriber" from "t102_case7" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'retain' flag to false...............................passed
And I set MQTT subscribe 'retain handling' property to "MQTT5_RETAIN_DO_NOT_SEND_AT_SUBSCRIPTION".passed
When I publish from "publisher" to "t102_case8" with qos 0 and message "Single not retained message in case8".passed
When I subscribe "subscriber" to "t102_case8" with qos 0....................passed
And message "Single not retained message in case8" is not received on "subscriber" from "t102_case8" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT subscribe 'retain as published' flag to false................passed
When I subscribe "subscriber" to "iot_data_0" with qos 0....................passed
And I set MQTT publish 'retain' flag to false...............................passed
And I set the 'retain' flag in expected received messages to false..........passed
When I publish from "publisher" to "iot_data_0" with qos 0 and message "Hello world1".passed
And message "Hello world1" received on "subscriber" from "iot_data_0" topic within 5 seconds.passed
And I set MQTT publish 'retain' flag to true................................passed
When I publish from "publisher" to "iot_data_0" with qos 0 and message "Hello world2".passed
And message "Hello world2" received on "subscriber" from "iot_data_0" topic within 5 seconds.passed
And I set MQTT subscribe 'retain as published' flag to true.................passed
When I subscribe "subscriber" to "iot_data_1" with qos 0....................passed
And I set MQTT publish 'retain' flag to false...............................passed
And I set the 'retain' flag in expected received messages to false..........passed
When I publish from "publisher" to "iot_data_1" with qos 0 and message "Hello world3".passed
And message "Hello world3" received on "subscriber" from "iot_data_1" topic within 5 seconds.passed
And I set MQTT publish 'retain' flag to true................................passed
And I set the 'retain' flag in expected received messages to true...........passed
When I publish from "publisher" to "iot_data_1" with qos 0 and message "Hello world4".passed
And message "Hello world4" received on "subscriber" from "iot_data_1" topic within 5 seconds.passed
And I add MQTT 'user property' with key "type" and value "json" to transmit.passed
And I add MQTT 'user property' with key "region" and value "Asia" to transmit.passed
And I add MQTT 'user property' with key "type" and value "json" to receive..passed
And I add MQTT 'user property' with key "region" and value "Asia" to receive.passed
When I subscribe "subscriber" to "iot_data_2" with qos 0....................passed
When I publish from "publisher" to "iot_data_2" with qos 0 and message "Expected userProperties are received".passed
And message "Expected userProperties are received" received on "subscriber" from "iot_data_2" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear MQTT 'user properties' to transmit..............................passed
And I clear MQTT 'user properties' to receive...............................passed
And I add MQTT 'user property' with key "agent-control" and value "control" to receive.passed
And I add MQTT 'user property' with key "timezone" and value "GMT6" to receive.passed
When I subscribe "subscriber" to "iot_data_2" with qos 0....................passed
When I publish from "publisher" to "iot_data_2" with qos 0 and message "Expected userProperties are not received".passed
And message "Expected userProperties are not received" is not received on "subscriber" from "iot_data_2" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear MQTT 'user properties' to transmit..............................passed
And I clear MQTT 'user properties' to receive...............................passed
And I add MQTT 'user property' with key "agent-control" and value "control" to transmit.passed
And I add MQTT 'user property' with key "timezone" and value "GMT6" to transmit.passed
When I subscribe "subscriber" to "iot_data_3" with qos 0....................passed
When I publish from "publisher" to "iot_data_3" with qos 0 and message "Ignore userProperties".passed
And message "Ignore userProperties" received on "subscriber" from "iot_data_3" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear MQTT 'user properties' to transmit..............................passed
And I clear MQTT 'user properties' to receive...............................passed
When I subscribe "subscriber" to "iot_data_4" with qos 0....................passed
When I publish from "publisher" to "iot_data_4" with qos 0 and message "Without userProperties".passed
And message "Without userProperties" received on "subscriber" from "iot_data_4" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to false.............passed
And I set the 'payload format indicator' flag in expected received messages to false.passed
When I subscribe "subscriber" to "payload_format_indicator_false_false" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_false_false" with qos 0 and message "Payload format indicators false/false".passed
And message "Payload format indicators false/false" received on "subscriber" from "payload_format_indicator_false_false" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to true..............passed
And I set the 'payload format indicator' flag in expected received messages to true.passed
When I subscribe "subscriber" to "payload_format_indicator_true_true" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_true_true" with qos 0 and message "Payload format indicators true/true".passed
And message "Payload format indicators true/true" received on "subscriber" from "payload_format_indicator_true_true" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to true..............passed
And I set the 'payload format indicator' flag in expected received messages to false.passed
When I subscribe "subscriber" to "payload_format_indicator_true_false" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_true_false" with qos 0 and message "Payload format indicators true/false".passed
And message "Payload format indicators true/false" is not received on "subscriber" from "payload_format_indicator_true_false" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to false.............passed
And I set the 'payload format indicator' flag in expected received messages to true.passed
When I subscribe "subscriber" to "payload_format_indicator_false_true" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_false_true" with qos 0 and message "Payload format indicators false/true".passed
And message "Payload format indicators false/true" is not received on "subscriber" from "payload_format_indicator_false_true" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to true..............passed
And I set the 'payload format indicator' flag in expected received messages to null.passed
When I subscribe "subscriber" to "payload_format_indicator_true_null" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_true_null" with qos 0 and message "Payload format indicators true/null".passed
And message "Payload format indicators true/null" received on "subscriber" from "payload_format_indicator_true_null" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT publish 'payload format indicator' flag to false.............passed
And I set the 'payload format indicator' flag in expected received messages to null.passed
When I subscribe "subscriber" to "payload_format_indicator_false_null" with qos 0.passed
When I publish from "publisher" to "payload_format_indicator_false_null" with qos 0 and message "Payload format indicators false/null".passed
And message "Payload format indicators false/null" received on "subscriber" from "payload_format_indicator_false_null" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT subscribe 'no local' flag to true............................passed
When I subscribe "subscriber" to "no_local_test" with qos 0.................passed
When I publish from "subscriber" to "no_local_true" with qos 0 and message "First message no local true test".passed
Then message "First message no local true test" is not received on "subscriber" from "no_local_true" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT subscribe 'no local' flag to false...........................passed
When I subscribe "subscriber" to "no_local_false" with qos 0................passed
When I publish from "subscriber" to "no_local_false" with qos 0 and message "First message no local false test".passed
Then message "First message no local false test" received on "subscriber" from "no_local_false" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT 'content type' to "text/plain; charset=utf-8" to transmit....passed
And I set MQTT 'content type' to "text/plain; charset=utf-8" to receive.....passed
When I subscribe "subscriber" to "content_types_the_same" with qos 0........passed
When I publish from "publisher" to "content_types_the_same" with qos 0 and message "Content types not null/not null".passed
And message "Content types not null/not null" received on "subscriber" from "content_types_the_same" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT 'content type' to "another content type value" to transmit...passed
And I set MQTT 'content type' to "text/plain; charset=utf-8" to receive.....passed
When I subscribe "subscriber" to "content_type_not_the_same" with qos 0.....passed
When I publish from "publisher" to "content_type_not_the_same" with qos 0 and message "Different values of content types".passed
And message "Different values of content type" is not received on "subscriber" from "content_type_not_the_same" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I clear MQTT 'content type' to transmit.................................passed
And I set MQTT 'content type' to "text/plain; charset=utf-8" to receive.....passed
When I subscribe "subscriber" to "content_type_null_not_null" with qos 0....passed
When I publish from "publisher" to "content_type_null_not_null" with qos 0 and message "Content types null/not null".passed
And message "Content types null/not null" is not received on "subscriber" from "content_type_null_not_null" topic within 5 seconds.passed
And I clear message storage.................................................passed
And I set MQTT 'content type' to "text/plain; charset=utf-8" to transmit....passed
And I clear MQTT 'content type' to receive..................................passed
When I subscribe "subscriber" to "content_type_not_null_null" with qos 0....passed
When I publish from "publisher" to "content_type_not_null_null" with qos 0 and message "Content types not null/null".passed
And message "Content types not null/null" received on "subscriber" from "content_type_not_null_null" topic within 5 seconds.passed
And I disconnect device "subscriber" with reason code 0.....................passed
And I disconnect device "publisher" with reason code 0......................passed
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 1]
```diff
@@ -104,6 +104,7 @@ public class MqttControlSteps {
 
     // publish default properties
     private static final boolean DEFAULT_PUBLISH_RETAIN = false;
+    private static final String DEFAULT_CONTENT_TYPE = null;
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:107]: Is that for publishing or expected to be received ?

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 2]
```diff
@@ -387,6 +394,48 @@ public void setMqttTimeoutSec(int mqttTimeoutSec) {
         log.info("MQTT timeout set to {} second(s)", mqttTimeoutSec);
     }
 
+    /**
+     * Sets MQTT content type to transmit.
+     *
+     * @param contentType MQTT content type to transmit
+     */
+    @And("I set MQTT 'content type' to {string} to transmit")
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Please use something similar to 
    @And("I set MQTT publish 'retain' flag to {booleanValue}")

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 3]
```diff
@@ -387,6 +394,48 @@ public void setMqttTimeoutSec(int mqttTimeoutSec) {
...
+        this.txContentType = contentType;
+        log.info("MQTT content type set to {} to transmit", contentType);
+    }
+
+    /**
+     * Sets MQTT content type to receive.
+     *
+     * @param contentType MQTT content type to receive
+     */
+    @And("I set MQTT 'content type' to {string} to receive")
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Please use something similar to 
    @And("I set the 'retain' flag in expected received messages to {booleanOrNullValue}")

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 4]
```diff
@@ -387,6 +394,48 @@ public void setMqttTimeoutSec(int mqttTimeoutSec) {
...
+    public void setMqttRxContentType(String contentType) {
+        this.rxContentType = contentType;
+        log.info("MQTT content type set to {} to receive", contentType);
+    }
+
+    /**
+     * Clear MQTT content type to transmit.
+     */
+    @And("I clear MQTT 'content type' to transmit")
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: Reorder code to  be updates txContentType together.
Rename the step.
Other steps has no 'to transmit' sentence in name

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java [hunk 5]
```diff
@@ -387,6 +394,48 @@ public void setMqttTimeoutSec(int mqttTimeoutSec) {
...
+    @SuppressWarnings("PMD.NullAssignment")
+    public void clearMqttTxContentType() {
+        this.txContentType = null;
+        log.info("MQTT content type reset to transmit");
+    }
+
+    /**
+     * Clear MQTT content type to receive.
+     */
+    @And("I clear MQTT 'content type' to receive")
```
Comment (bgklika) [inline uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: I reset MQTT 'content type' in expected received messages

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 1 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java:107]: this is the default value for both of them.

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 2 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: The step was renamed in 41edf65c896419d5e3fc6a8684a8ffd6126319b9

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 4 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: The step was renamed in 41edf65c896419d5e3fc6a8684a8ffd6126319b9

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 5 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: The step was renamed in 41edf65c896419d5e3fc6a8684a8ffd6126319b9

File: uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java (hunk 3 above)
Comment (auarbekov) [reply uat/testing-features/src/main/java/com/aws/greengrass/steps/MqttControlSteps.java]: The step was renamed in 41edf65c896419d5e3fc6a8684a8ffd6126319b9

**Review by eschastlivtsev — APPROVED** [review]

**Review by bgklika — APPROVED** [review]

**Pull request closed without merging**